x-pack/plugins/security_solution/public/management/pages/trusted_apps/service/to_new_trusted_app.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { MaybeImmutable, NewTrustedApp } from '../../../../../common/endpoint/types';
+import { defaultNewTrustedApp } from '../store/builders';
+
+const NEW_TRUSTED_APP_KEYS: Array<keyof NewTrustedApp> = [
+  'name',
+  'effectScope',
+  'entries',
+  'description',
+  'os',
+];
+
+export const toNewTrustedApp = <T extends NewTrustedApp>(
+  trustedApp: MaybeImmutable<T>
+): NewTrustedApp => {
+  const newTrustedApp = defaultNewTrustedApp();
+  for (const key of NEW_TRUSTED_APP_KEYS) {
+    // This should be safe. Its needed due to the inter-dependency on property values (`os` <=> `entries`)
```

Comment:
can you maybe go over this with me? i feel like i might need this for my protections form refactor

Reply:
"should be safe" <- famous last words?